src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -387,13 +414,57 @@ class MediaBaseViewer extends BaseViewer {
      */
     togglePlay() {
         if (this.mediaEl.paused) {
-            this.mediaEl.play();
-            this.emit('play');
-            this.handleRate();
-            this.handleVolume();
+            this.play();
         } else {
-            this.mediaEl.pause();
-            this.emit('pause');
+            this.pause();
+        }
+    }
+
+    /**
+     * Validates time parameter
+     *
+     * @private
+     * @param {number} time - time for media
+     * @return {boolean} - true if time is valid
+     */
+    isValidTime(time) {
+        return !isNaN(time) && isFinite(time) && time >= 0 && time <= this.mediaEl.duration;
+    }
+
+    /**
+     * Pauses segment at certain time
+     *
+     * @private
+     * @param {number} time - time for media
+     * @emits pause
+     * @return {void}
+     */
+    pauseSegmentAt(time) {
+        // Remove earlier event listener
+        this.mediaEl.removeEventListener('timeupdate', this.pauseSegmentListener);
+        this.pauseSegmentListener = () => {
+            if (this.mediaEl.currentTime > time) {
+                this.pause();
+            }
+        };
+        this.mediaEl.addEventListener('timeupdate', this.pauseSegmentListener);
+    }
+
+    /**
+     * Play segment from start time to end time
+     *
+     * @private
+     * @param {number} start - start time in seconds
+     * @param {number} end - end time in seconds
+     * @return {void}
+     */
+    playSegment(start, end) {
```

Comment:
Hmm, don't think we should add a different API for playing segments to keep things simple.
We can get away with just 1 `play()` function, that optionally takes `start` and `end`.

play() ---> what it does today, plays
play(23) ---> plays starting 23sec if 23 is a number
play(12, 23) ---> plays starting 12 and pauses at 23

So, something like:

```js
play(start, end) {
  const hasValidStart = this.isValidTime(start);
  const hasValidEnd = this.isValidTime(end);

  if (hasValidStart) {
    this.setMediaTime(start);
    if (hasValidEnd) {
      this.pauseSegmentAt(end);
    }
  }
  
   // code for play
}
```

Likewise I think pause function can be combined.

```js
pause(time) {
  const hasValidTime = this.isValidTime(time);

  // Remove event listener

  if (hasValidTime) {
    // similar code you have for pauseSegmentAt(time)
  }
  
   // code for pause
}
```